airflow/models/taskinstance.py

```diff
@@ -436,6 +436,7 @@ class TaskInstance(Base, LoggingMixin):
     next_method = Column(String(1000))
     next_kwargs = Column(MutableDict.as_mutable(ExtendedJSON))
 
+    _task_display_name = Column("task_display_name", String(2000), default="")
```

Comment:
Should this have `nullable=False`?

Reply:
The model is defined as far as I understand the code to use the `task_id` if no display name is given. This is good as it saved space in the DB (if no display name given) as well as allows migration (else during migration the column needs to be filled).
I'd propose to keep it this way.

Reply:
> The model is defined as far as I understand the code to use the `task_id` if no display name is given. This is good as it saved space in the DB (if no display name given) as well as allows migration (else during migration the column needs to be filled).
> I'd propose to keep it this way.

Do you mean we should allow the DB colomn Nullable and fallback to `task_id` if no display name is null (None in Python)? 

What if display_name is an empty string but not None, should we fallback to `task_id` or use empty string as it?


@uranusjr What do you think?

Reply:
> Do you mean we should allow the DB colomn Nullable and fallback to `task_id` if no display name is null (None in Python)?
> 

Yes.

I hope nobody sets an empty string as display, but if we want to get into a strong data validation (e.g. display need to be at least one character in length) we can validate in python at point of dag parsing before it gets into DB.

Reply:
I don’t quite understand why we should disinguish between an empty string and None. The default display name in the DAG interface is an empty string, so using that to signify _no display name_ and falling back to dag_id/task_id makes sense to me without any additional validation. Adding None to the mix seems unnecessarily complex.

Reply:
I understand your point. 
I think @jens-scheffler-bosch is considering from the aspect of database storage.

Actually I'm ok with either
1) make it nullable, or
2) make it not nullable with default "" (empty string)

I think both proposals above should fallback to task_id/dag_id when display name is empty/null.

I personally prefer 1) which is my original change because it's more friendly to migrations. But they're not too much different and I'm perfectly ok with 2.

Let's try to make a decision and move forward.

Reply:
Personally I prefer to keep the database tidy while sacraficing some migration simplicity because migration only happens once, while we are stuck with the database schema from there onwards.

Reply:
From my DBA legacy (with MSSQL) I know that NULL needs less storage than an empty string, don't know how it is with Postgres/MYSQL but I assume keeping it NULL per default is saving storage space. So I also would prefer (1) (plus it is safer, in cases somebody badly tries to check if `isinstance(db_value, str)` for an empty string :-D

Reply:
Can we reach an agreement here so I can change accordingly. It's not a big deal IMO.

Reply:
If the majority sees it different I don't want to block this PR and would favor to move ahead. But for me an "empty"==undefined display value always corresponds with NULL, not an empty string. Alongside thing about if you add the DB Column during migration, the field per default will be set to NULL as well. If you want to migrate existing tables making an empty stringas default is another operation before being able to set a NOT NULL constraint.

In summay no major concerns but not convinced having an empty string as default (I see no benefits just overhead).

Reply:
But as I take a look to current code, it is using NULL anyway, so all is good - LGTM!